packages/network/src/network.js

```diff
@@ -14,6 +15,125 @@ import '@agoric/store/exported.js';
  */
 export const ENDPOINT_SEPARATOR = '/';
 
+const Shape1 = /** @type {const} */ ({
+  /**
+   * Data is string | Buffer | ArrayBuffer
+   * but only string is passable
+   */
+  Data: M.string(),
+  Bytes: M.string(),
+  Endpoint: M.string(),
+  // XXX avoid: cannot check unrecognized tag "Vow": "[Vow]"
+  Vow: M.any(),
+
+  ConnectionHandler: M.remotable('ConnectionHandler'),
+  Connection: M.remotable('Connection'),
+  InboundAttempt: M.remotable('InboundAttempt'),
+  Listener: M.remotable('Listener'),
+  ListenHandler: M.remotable('ListenHandler'),
+  Port: M.remotable('Port'),
+  ProtocolHandler: M.remotable('ProtocolHandler'),
+  ProtocolImpl: M.remotable('ProtocolImpl'),
+});
+
+const Shape2 = /** @type {const} */ ({
+  ...Shape1,
+  Vow$: shape => M.or(shape, Shape1.Vow),
+  AttemptDescription: M.splitRecord(
+    { handler: Shape1.ConnectionHandler },
+    { remoteAddress: Shape1.Endpoint, localAddress: Shape1.Endpoint },
+  ),
+  Opts: M.recordOf(M.string(), M.any()),
+});
+
+export const Shape = /** @type {const} */ harden({
+  ...Shape2,
+  ConnectionI: M.interface('Connection', {
+    send: M.callWhen(Shape2.Data)
+      .optional(Shape2.Opts)
+      .returns(Shape2.Vow$(Shape2.Bytes)),
+    close: M.callWhen().returns(Shape2.Vow$(M.undefined())),
+    getLocalAddress: M.call().returns(Shape2.Endpoint),
+    getRemoteAddress: M.call().returns(Shape2.Endpoint),
+  }),
+  InboundAttemptI: M.interface('InboundAttempt', {
+    accept: M.callWhen(Shape2.AttemptDescription).returns(
+      Shape2.Vow$(Shape2.Connection),
+    ),
+    getLocalAddress: M.call().returns(Shape2.Endpoint),
+    getRemoteAddress: M.call().returns(Shape2.Endpoint),
+    close: M.callWhen().returns(Shape2.Vow$(M.undefined())),
+  }),
+  PortI: M.interface('Port', {
+    getLocalAddress: M.call().returns(Shape2.Endpoint),
+    addListener: M.callWhen(Shape2.Listener).returns(
+      Shape2.Vow$(M.undefined()),
+    ),
+    connect: M.callWhen(Shape2.Endpoint)
+      .optional(Shape2.ConnectionHandler)
+      .returns(Shape2.Vow$(Shape2.Connection)),
+    removeListener: M.callWhen(Shape2.Listener).returns(
+      Shape2.Vow$(M.undefined()),
+    ),
+    // @@ types.js says returns void; code says string
```

Comment:
Should we have it return a string in `type.js` as well? Given that all the current callers (in our codebase) treat it like a void, it shouldn't be a breaking change

Reply:
@michaelfig I meant to check with you on this. help?

(@@ is my marker for "do not merge", fwiw. I should probably just spell it out)

Reply:
Thanks for asking!

My preference is to leave the TS return type as `void` (or PromiseVow<void> if needed), and to change `revoke()`'s implementation to return undefined (implicitly or explicitly).  Finally, specify the methodGuard's return value here as an `M.undefined()` (or however it needs to be wrapped).